File: rust/pn.rs
fn main() {
    let mut p = true;
    let n: u32 = 100000;
    println!("2\n3");

    for i in (5..n).step_by(2) {
        for j in (3..=(i / 2)).step_by(2) {
            if i % j == 0 {
                p = false;
                break;
            } else {
                p = true;
            }
        }

        if p {
            println!("{}", i);
            p = false;
        }
    }
}

File: zig/pn.zig
const std = @import("std");
const print = std.io.getStdOut().writer().print;

pub fn main() !void {
    var p = true;
    const n: u32 = 100000;
    try print("2\n3\n", .{});

    var i: usize = 5;
    while (i < n):(i += 2) {
        var j: usize = 3;
        while (j <= @as(usize, i / 2)):(j += 2) {
            if (i % j == 0) {
                p = false;
                break;
            } else {
                p = true;
            }
        }

        if (p) {
            try print("{}\n", .{i});
            p = false;
        }
    }
}
